crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs

```diff
@@ -132,13 +137,18 @@ impl DataFlowGraph {
     }
 
     /// Inserts a new instruction at the end of the given block and returns its results
+    ///
+    /// TODO: This is doing more than just inserting an instruction and its results
```

Comment:
Sounds like the missing piece is that this method attempts to simplify the instruction to another value and only inserts it if it could not do so. We should:

1. Update the doc comment above to note this.
2. Potentially rename the function? I'm unsure what a clearer name would be.